magicblock-committor-service/src/intent_executor/intent_executor_factory.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+use magicblock_rpc_client::MagicblockRpcClient;
+use magicblock_table_mania::TableMania;
+
+use crate::{
+    intent_executor::{
+        commit_id_fetcher::CommitIdTrackerImpl, IntentExecutor,
+        IntentExecutorImpl,
+    },
+    transaction_preperator::transaction_preparator::TransactionPreparatorV1,
+    ComputeBudgetConfig,
+};
+
+pub trait IntentExecutorFactory {
+    type Executor: IntentExecutor;
+
+    fn create_instance(&self) -> Self::Executor;
+}
+
+/// Dummy struct to simplify signature of CommitSchedulerWorker
+pub struct IntentExecutorFactoryImpl {
+    pub rpc_client: MagicblockRpcClient,
+    pub table_mania: TableMania,
+    pub compute_budget_config: ComputeBudgetConfig,
+    pub commit_id_tracker: Arc<CommitIdTrackerImpl>,
+}
+
+impl IntentExecutorFactory for IntentExecutorFactoryImpl {
```

Comment:
Not sure if we need a full blown factory here. It's a rarely used pattern in Rust Builders are more common, but this is so simple that just adding a `new` method to `IntentExecutorImpl` seems sufficient no?

You can even have type narrowed implementations if you want to.

This here smells too much like C++/Java to me tbh :)